app/src/main/java/ch/epfl/sweng/rps/persistance/Cache.kt

```diff
@@ -0,0 +1,67 @@
+package ch.epfl.sweng.rps.persistance
+
+import android.content.Context
+import ch.epfl.sweng.rps.db.FirebaseHelper
+import ch.epfl.sweng.rps.db.FirebaseReferences
+import ch.epfl.sweng.rps.db.FirebaseRepository
+import ch.epfl.sweng.rps.models.User
+import ch.epfl.sweng.rps.models.UserStat
+import java.net.InetAddress
+
+
+class Cache private constructor(val ctx:Context, val preferFresh:Boolean = false) {
+    companion object {
+        var cache:Cache? = null
+        fun getInstance(ctx:Context, preferFresh: Boolean = false): Cache {
+            if(cache != null) return cache!!
+            cache = Cache(ctx, preferFresh)
+            return cache!!
+        }
+    }
+    private val fbRef = FirebaseReferences()
+    private val fbRepo = FirebaseRepository.createInstance(fbRef)
```

Comment:
Ideally you should use `ServiceLocator` here and not instanciate the classes by yourself.

Reply:
Yes , I could also modify the firebaseRepository to be singleton only like I do for cache

Reply:
It's already a singleton, when accessed via ServiceLocator.

Reply:
Now it is accessed via service locator and the repo injection is completely transparent to developers! 👍